src/ttsim/plot/dag/tt.py

```diff
@@ -2,109 +2,131 @@
-from dataclasses import dataclass
-from typing import TYPE_CHECKING, Literal
+from typing import TYPE_CHECKING, Any, Literal
 
-import dags
 import dags.tree as dt
 import networkx as nx
 
 from ttsim import main
-from ttsim.plot.dag.shared import NodeMetaData, dummy_callable, get_figure
+from ttsim.interface_dag_elements.fail_if import format_errors_and_warnings
+from ttsim.main_args import OrigPolicyObjects, TTTargets
+from ttsim.main_target import MainTarget
+from ttsim.plot.dag.shared import NodeMetaData, get_figure
 from ttsim.tt import (
     ParamFunction,
     ParamObject,
-    PolicyInput,
+    TimeConversionFunction,
 )
 
 if TYPE_CHECKING:
     from pathlib import Path
 
     import plotly.graph_objects as go
 
+    from ttsim.main_args import InputData, Labels
     from ttsim.typing import (
+        DashedISOString,
         PolicyEnvironment,
+        QNameData,
         SpecEnvWithoutTreeLogicAndWithDerivedFunctions,
     )
 
 
-@dataclass(frozen=True)
-class NodeSelector:
-    """Select nodes from the DAG."""
-
-    node_paths: list[tuple[str, ...]]
-    type: Literal["neighbors", "descendants", "ancestors", "nodes"]
-    order: int | None = None
-
-
-@dataclass(frozen=True)
-class _QNameNodeSelector:
-    """Select nodes from the DAG."""
-
-    qnames: list[str]
-    type: Literal["neighbors", "descendants", "ancestors", "nodes"]
-    order: int | None = None
-
-
 def tt(
-    policy_date_str: str,
+    *,
+    # Args specific to TTSIM plotting
     root: Path,
-    node_selector: NodeSelector | None = None,
-    title: str = "",
+    selection_type: Literal["neighbors", "descendants", "ancestors", "nodes"]
+    | None = None,
+    selection_depth: int | None = None,
     include_params: bool = True,
     show_node_description: bool = False,
     output_path: Path | None = None,
+    # Elements of main
+    policy_date_str: DashedISOString | None = None,
+    orig_policy_objects: OrigPolicyObjects | None = None,
+    input_data: InputData | None = None,
+    tt_targets: TTTargets | None = None,
+    processed_data: QNameData | None = None,
+    labels: Labels | None = None,
+    policy_environment: PolicyEnvironment | None = None,
+    backend: Literal["numpy", "jax"] = "numpy",
+    include_fail_nodes: bool = True,
+    include_warn_nodes: bool = True,
+    # Args specific to plotly
+    **kwargs: Any,  # noqa: ANN401
 ) -> go.Figure:
     """Plot the TT DAG.
 
     Parameters
     ----------
-    policy_date_str
-        The date string.
     root
         The root path.
-    node_selector
-        The node selector. Default is None, i.e. the entire DAG is plotted.
-    title
-        The title of the plot.
+    selection_type
+        The type of the DAG to plot. Can be one of:
+        - "neighbors": Plot the neighbors of the target nodes.
+        - "descendants": Plot the descendants of the target nodes.
+        - "ancestors": Plot the ancestors of the target nodes.
+        - "nodes": Plot the target nodes.
```

Comment:
You mentioned it in your edit, but I truly have no idea what is supposed to happen for "nodes".